### PR TITLE
Merge 1.4 1.5

### PR DESCRIPTION
**Pull request by solivier**

### CODE REVIEW

File: features/Context/Page/Product/Edit.php [hunk 1]
```diff
@@ -656,6 +656,29 @@ public function checkCompletenessMissingValues($channelCode, $localeCode, $info)
...
+     * @param string $missingValue
+     * @param string $expectedLabel
+     *
+     * @throws ExpectationException
+     */
+    public function checkCompletenessMissingValuesLabels($channelCode, $localeCode, $missingValue, $expectedLabel)
+    {
+        $completenessCell = $this->spin(function () use ($channelCode, $localeCode) {
+            return $this->findCompletenessCell($channelCode, $localeCode)->find('css', '.missing');
+        }, 20, 'Unable to find completenesses cell');
```
Comment (grena) [inline features/Context/Page/Product/Edit.php]: Remove timeout param please

File: features/Context/Page/Product/Edit.php [hunk 2]
```diff
@@ -656,6 +656,29 @@ public function checkCompletenessMissingValues($channelCode, $localeCode, $info)
...
+
+        $missingValueElement = $this->spin(function () use ($completenessCell, $missingValue) {
+            return $completenessCell->find('css', sprintf('span[data-attribute="%s"]', $missingValue));
+        }, 20, 'Unable to find completenesses missing value element');
```
Comment (grena) [inline features/Context/Page/Product/Edit.php]: Remove timeout param please

File: features/Context/WebUser.php [hunk 3]
```diff
@@ -850,7 +850,9 @@ public function iChangeTheTo($field, $value = null, $language = null)
     {
         if (null !== $language) {
             try {
-                $field = $this->getCurrentPage()->getFieldLocator($field, $this->getLocaleCode($language));
+                $field = $this->spin(function () use ($field, $language) {
+                    return $this->getCurrentPage()->getFieldLocator($field, $this->getLocaleCode($language));
+                }, 30);
```
Comment (grena) [inline features/Context/WebUser.php]: Remove timeout param please

Comment (skeleton) [inline features/Context/WebUser.php]: remove 30

File: features/Context/WebUser.php [hunk 4]
```diff
@@ -1976,6 +1977,34 @@ public function iShouldSeeTheCompleteness(TableNode $table)
...
+     * @Then /^I should see the missing completeness labels:$/
+     *
+     * @throws ExpectationException
+     */
+    public function iShouldSeeTheMissingCompletenessLabels(TableNode $table)
+    {
+        $page = $this->getCurrentPage();
+        $collapseSwitchers = $this->spin(function () use ($page) {
+            return $page->findAll('css', '.completeness-block header .btn');
+        }, 20, 'Unable to find completenesses block');
```
Comment (skeleton) [inline features/Context/WebUser.php]: remove 20

Comment (grena) [inline features/Context/WebUser.php]: Remove timeout param please

File: src/Pim/Bundle/CatalogBundle/Manager/CompletenessManager.php [hunk 5]
```diff
@@ -149,7 +149,7 @@ public function scheduleForFamily(FamilyInterface $family)
      *      ],
      * ]
      */
-    public function getProductCompleteness(ProductInterface $product, array $channels, array $locales, $localeCode)
+    public function getProductCompleteness(ProductInterface $product, array $channels, array $locales, $localeCode = null)
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Manager/CompletenessManager.php]: should be rewritten I guess (maybe not in this PR)

Comment (grena) [inline src/Pim/Bundle/CatalogBundle/Manager/CompletenessManager.php]: Still needed? ping @philippemossiere

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js [hunk 6]
```diff
@@ -87,7 +87,7 @@ function(
          *
          * @property
          */
-        dateWidgetSelector: '.datepicker',
+        dateWidgetSelector: 'div.datepicker',
```
Comment (skeleton) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: no. `div` has been removed in 1.5

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js [hunk 7]
```diff
@@ -171,8 +171,8 @@ function(
             }
 
             if (value.value) {
-                start = value.value.start;
-                end   = value.value.end;
+                start = value.value.start.format;
+                end   = value.value.end.format;
```
Comment (skeleton) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: same, `format` has been removed in 1.5

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js (hunk 6 above)
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: right

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js (hunk 7 above)
Comment (phaseinducer) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: absolutely

**Pull request merged**